snapshot/src/androidMain/kotlin/com/quickbird/snapshot/Color.kt

```diff
@@ -1,8 +1,117 @@
 package com.quickbird.snapshot
 
+import android.graphics.Color as AndroidColor
 import androidx.annotation.ColorInt
+import kotlin.collections.component1
+import kotlin.math.abs
+import kotlin.math.cbrt
+import kotlin.math.min
+import kotlin.math.pow
+import kotlin.math.sqrt
 
 data class Color(@ColorInt val value: Int)
 
 val @receiver:ColorInt Int.color
     get() = Color(this)
+
+/**
+ * Calculates [Delta E (1994)](http://zschuessler.github.io/DeltaE/learn/#toc-delta-e-94) between
+ * two colors in the CIE LAB color space returning a value between 0.0 - 1.0 (0.0 means no difference, 1.0 means completely opposite)
+ */
+fun Color.deltaE(other: Color): Double {
+    if (this == other) {
+        return 0.0
+    }
+    // Delta E (1994) is in a 0-100 scale, so we need to divide by 100 to transform it to a percentage
+    //
+    return min(this.deltaE1994(other) / 100, 1.0)
+}
+
+/**
+ * Convert the color to the CIE XYZ color space within nominal range of [0.0, 1.0]
+ * using sRGB color space and D65 white reference white
+ */
+private fun Color.toXYZ(): DoubleArray {
+    // Values must be within nominal range of [0.0, 1.0]
+    //
+    var r = AndroidColor.red(this.value) / 255.0
+    var g = AndroidColor.green(this.value) / 255.0
+    var b = AndroidColor.blue(this.value) / 255.0
+
+    // Inverse sRGB companding
+    //
+    r = if (r <= 0.04045) r / 12.92 else ((r + 0.055) / 1.055).pow(2.4)
+    g = if (g <= 0.04045) g / 12.92 else ((g + 0.055) / 1.055).pow(2.4)
+    b = if (b <= 0.04045) b / 12.92 else ((b + 0.055) / 1.055).pow(2.4)
+
+    // Linear RGB to XYZ using sRGB color space and D65 white reference white
+    //
+    return doubleArrayOf(
+        0.4124564 * r + 0.3575761 * g + 0.1804375 * b,
+        0.2126729 * r + 0.7151522 * g + 0.0721750 * b,
+        0.0193339 * r + 0.1191920 * g + 0.9503041 * b
+    )
+}
+
+/**
+ * Convert the color to the CIE LAB color space using sRGB color space and D65 white reference white
+ */
+private fun Color.toLAB(): DoubleArray {
+    val (x, y, z) = this.toXYZ()
+
+    // CIE standard illuminant D65
+    //
+    val Xr = 0.9504
+    val Yr = 1.000
+    val Zr = 1.0888
+
+    val xr = x / Xr
+    val yr = y / Yr
+    val zr = z / Zr
+
+    val e = 0.008856
+    val k = 903.3
+
+    val fx = if (xr > e) cbrt(xr) else ((k * xr) + 16) / 116.0
+    val fy = if (yr > e) cbrt(yr) else ((k * yr) + 16) / 116.0
+    val fz = if (zr > e) cbrt(zr) else ((k * zr) + 16) / 116.0
+
+    return doubleArrayOf(
+        116 * fy - 16,
+        500 * (fx - fy),
+        200 * (fy - fz)
+    )
+}
```

Comment:
Thank you for your contribution, from API LEVEL 26 [ColorSpace API](https://developer.android.com/reference/android/graphics/ColorSpace) is available and I would suggest using this and raising the min-sdk level to 26, we are fine with supporting from Android O onwards

What do you think @JoelWhitney ?

```suggestion
/**
 * Convert the color to the CIE LAB color space using sRGB color space and D65 white reference white
 */
private fun Color.toLAB(): FloatArray {
    val labConnector = ColorSpace.connect(
        ColorSpace.get(ColorSpace.Named.SRGB),
        ColorSpace.get(ColorSpace.Named.CIE_LAB)
    )

    val rgb = floatArrayOf(
        AndroidColor.red(value) / 255.0f,
        AndroidColor.green(value) / 255.0f,
        AndroidColor.blue(value) / 255.0f
    )

    return labConnector.transform(rgb[0], rgb[1], rgb[2])
}
```

Reply:
Looks good to me! I had some tests in another branch (since I had to update gradle version) and I confirmed all of the tests still pass with these changes

Reply:
Would be nice if you could add the the tests here too.
i will merge this now. 

thank you